use-cases/xtclim/src/anomaly.py

```diff
@@ -1,114 +1,182 @@
-            test_avg_losses += test_avg_loss
-        tot_train_losses = np.array(tot_train_losses)/n_avg
-        tot_test_losses = np.array(tot_test_losses)/n_avg
-        train_avg_losses = train_avg_losses/n_avg
-        test_avg_losses = test_avg_losses/n_avg
-        
-        pd.DataFrame(tot_train_losses).to_csv(f"../outputs/train_losses_{season}1d_allssp.csv")
-        pd.DataFrame(tot_test_losses).to_csv(f"../outputs/test_losses_{season}1d_allssp.csv")
-        print('Train average loss:', train_avg_losses)
-        print('Test average loss:', test_avg_losses)
-
-
-if future_evaluation:
-    for season in ['winter_', 'spring_', 'summer_', 'autumn_']:
-        
-        # load previously trained model
-        cvae_model = model.ConvVAE().to(device)
-        cvae_model.load_state_dict(torch.load(f'../outputs/cvae_model_{season}1d.pth'))
-
-        for scenario in ['585', '370', '245', '126']:
-            
-            # projection set and data loader
-            proj_time = pd.read_csv(f"../input/dates_proj_{season}data.csv")
-            proj_data = np.load(f"../input/preprocessed_1d_proj{scenario}_{season}data_allssp.npy")
-            n_proj = len(proj_data)
-            projset = [ ( torch.from_numpy(np.reshape(proj_data[i], (3, 32, 32))), 
-                            proj_time['0'][i] ) for i in range(n_proj) ]
-            projloader = DataLoader(
-                projset, batch_size=1, shuffle=False
+        self.evaluation = evaluation
+        self.batch_size = batch_size
+        self.seasons = seasons
+        self.config = TrainingConfiguration(
+            batch_size=batch_size,
+        )
+
+    @monitor_exec
+    def execute(self):
+        # Initialize distributed backend
+        self._init_distributed_strategy()
+        if self.evaluation=='past':
+            season = self.seasons
+            # load previously trained model
+            cvae_model = NNmodel.ConvVAE().to(self.device)
+            cvae_model.load_state_dict(self.model['model_state_dict'], strict=False)
+
+            distribute_kwargs = {}
+            # Distributed model, optimizer, and scheduler
+            cvae_model, _, _ = self.strategy.distributed(
+                cvae_model, None, None, **distribute_kwargs
             )
 
-            # get the losses for each data set
-            # on various experiments to have representative statistics
-            proj_avg_losses, _, tot_proj_losses, _ = evaluate(cvae_model, projloader, 
-                                                        projset, device, criterion, 
-                                                        pixel_wise_criterion)
-            
+            # train set and data loader
+            train_time = pd.read_csv(f"input/dates_train_{season}data_1memb.csv")
+            train_data = np.load(f"input/preprocessed_1d_train_{season}data_1memb.npy")
+            n_train = len(train_data)
+            trainset = [ ( torch.from_numpy(np.reshape(train_data[i], (2, 32, 32))),
```

Comment:
You should not have spaces after opening brackets. You should also make a newline after the opening brackets if the line is too long. See earlier comments about this.